### ISSUE TITLE
[Discussion] Repo use cases and implementation

### ISSUE THREAD

**Issue by GitBytes** [pacifica/ansible-travis#20]
Want to open a dialog here to better nail down the use case for this role and discuss its implementation.

So, first off we (Pacifica) have been developing and using the [pacifica/ansible-pacifica](https://github.com/pacifica/ansible-pacifica) role, which is generally accepted as the role to be used to deploy Pacifica services in production environments.

This repo however, seems to be something of a clone of that role but is drastically different and I am not sure why it is being implemented in this way. If it were an example repo to show different deployment tests, it _should_ be consuming the other role and testing the limits of that role to do what this role is intending (i.e. making ansible-pacifica a direct dependency of this role).

Having this role out there not consuming the other and duplicating code but also not is really confusing to current and future users IMHO, and we should either change how this repo is being utilized OR remove/combine the roles. I am a bit on the fence on which might be the cleaner choice at the moment but I am hoping this discussion can clear up that point.

**Comment by dmlb2000** [comment]
The purpose of the role is to deploy Pacifica using the https://github.com/pacifica/ansible-pacifica role in Travis CI environments. There is no duplication of the Pacifica ansible role. This role consumes the Pacifica role (https://github.com/pacifica/ansible-travis/blob/master/requirements.yml) and appropriately configures Pacifica services in Travis CI for any number of versions of Python and specific dependent infrastructure for individual Pacifica repositories.

So the goal is to replace the complicated testing dependencies for each Pacifica service with a simple set of Ansible commands to configure Travis. https://github.com/pacifica/pacifica-notifications/commit/c22d3d8d4dd82331b8a49589b60b215dfb3d0332

**Comment by GitBytes** [comment]
@dmlb2000 fair enough. Somehow I did missed the `requirements.yml` where you specified consuming the main pacifica role. I think I was looking for dependencies in the meta for the role and didn’t think of this as a playbook but a role. If this is more of a playbook then there are some merging of concepts being seen here that might be muddying the waters for me a bit. If it is a role it should be structured as such. If it is a "project" that consumes roles then things like meta are not necessary.

I see the use case described and I see the requirements.yml but from what I can tell it is a non-standard practice to use molecule the way it is being used as molecule is designed to test roles and not "non-roles". Not to say it isn't an interesting way to apply molecule, it is just non-standard and feels weird in implementation. I cannot tell if this repo is an "ansible role" (i.e. to be consumed by others) or an "ansible project" (i.e. consumer of roles), which should generally be considered two separate constructs when using ansible. I think bridging these two concepts into a singular repo structure is confusing and against best practices. Happy to field docs or other examples that may prove otherwise.

Proposed solution:
* Break out the "role" bits into role(s) in new repos with their own testing
* Consume both the travis/ci/testing role (and dependencies) in an ansible project (this repo?) with playbooks or a site.yml that aggregates the calls for the roles and runs them in a pipeline that may not include the use of molecule specifically as the most benefit from molecule is gained when testing roles, not projects.

**Comment by GitBytes** [comment]
@dmlb2000 In addition, I would not consider this repo simple by any means. Trying to read through this and trace what is happening when and where is a bit..tough... Just my 2 cents as I was trying to pry this all apart to figure out what was going on.

**Comment by GitBytes** [comment]
I also see the notifications commit that consumes this as part of other travis pipelines but I still think this could be simplified and consumed as a project with essentially just simple playbooks for each "scenario" (not in the molecule sense) rather than this awkward role structure with molecule someplace in the middle..

**Comment by dmlb2000** [comment]
It is complicated yes. The goal was to configure the require dependencies for testing any Pacifica repository in Travis with three lines of shell

```
pip install ansible
ansible-galaxy install ...
ansible-playbook ...
```

If you have thoughts on how to achieve that in a better way, it would be nice to see the repository structure to achieve the goal and we can refactor from there.

**Comment by GitBytes** [comment]
I already forked and will lean in on helping achieve this. I agree it is needed for the use case you have described but I just want to be clear on goals so I don't go down a rabbit hole and waste time and effort. I think I have an idea of what we can do based on my descriptions and discussion so far so I'll try to get something put together this week.
